tensorly/backend/core.py

```diff
@@ -0,0 +1,689 @@
+import inspect
+import importlib
+import os
+import sys
+import threading
+import types
+import warnings
+
+import numpy as np
+import scipy.linalg
+import scipy.sparse.linalg
+
+
+class Backend(object):
+    @classmethod
+    def register_method(cls, name, func):
+        """Register a method with the backend.
+
+        Parameters
+        ----------
+        name : str
+            The method name.
+        func : callable
+            The method
+        """
+        setattr(cls, name, staticmethod(func))
+
+    @property
+    def int64(self):
+        raise NotImplementedError
```

Comment:
I think this should be `NotImplementedError()` (note the extra parantheses). We're raising an instance of the class, not the class itself.

I would much prefer that we use the [`abstractmethod` decorator](https://docs.python.org/3.7/library/abc.html#abc.abstractmethod) here instead, and made `Backend` into an [`ABC`](https://docs.python.org/3.7/library/abc.html#abc.ABC). However, I'm fine with just raising an error as well.